sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -1076,6 +1076,56 @@ struct _pi_queue : _pi_object {
 
   // Indicates that the queue is healthy and all operations on it are OK.
   bool Healthy{true};
+
+  // Below are data structures and methods for optimizing events usage in scope
+  // of the same queue. We have more information for events reuse within the
+  // same in-order queue. An event can be reused on the queue even before it is
+  // synchronized because prior commands in the same queue guarantee its
+  // completion. I.e. as soon as some command is submitted to the in-order queue
+  // then it means that it depends on the previous command and the event
+  // associated with the previous command can be reset and reused for the next
+  // command submission:
+  //    previous command -> Event1
+  //    command (depends on Event1) -> Event2
+  //    next command (depends on Event2) -> (reset and reuse Event1 here)
+  // Such dependency chain is maintained only in scope of the same in-order
+  // queue. To be able to reuse event across queues we need to wait when event
+  // is completed and its external references are gone which almost always means
+  // that event is ready to be destroyed and we already have context-level cache
+  // of events for that.
+
+  // Copy of the last command event which is suitable for reuse.
+  // It will be put into the cache when new command is submitted to the in-order
+  // queue.
+  pi_event CandidateForReuse = nullptr;
+
+  // Caches of events for reuse.
+  std::vector<std::list<pi_event>> EventCaches{2};
```

Comment:
why no need to choose if event is with or without profiling?

Reply:
Because profiling enabled/disabled is the property of the whole queue.